File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    // array의 commands[i][0]부터 commands[i][1]까지 자르기
    for(int i=0; i<commands.size(); i++) {
        vector<int> result;
        for(int j=commands[i][0]-1; j<commands[i][1]; j++) {
            result.push_back(array[j]);
        }
        sort(result.begin(), result.end());
        int number = result[commands[i][2]-1];
        answer.push_back(number);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        // 1.commands[i][0] ~ commands[i][1] 자르기
        // 2.오름차순 정렬 
        // 3.commands[i][2] 번째 수 고르기
        for(int i=0; i<commands.length; i++) {
            int[] res = Arrays.copyOfRange(array, commands[i][0]-1, commands[i][1]);
            Arrays.sort(res);
            answer[i] = res[commands[i][2]-1];
        }
        
        return answer;
    }
}